fix(storybook): voeg subpath export toe voor icon-registry.generated

## User Story

Als ontwikkelaar wil ik dat `tsc --noEmit` geen TS2307-fouten geeft op de `@dsn/components-react/icon-registry.generated` import, zodat de TypeScript check betrouwbare output geeft.

## Context

Meerdere story-bestanden importeren het `IconName` type via een subpath van het `@dsn/components-react` pakket:

\`\`\`ts
import type { IconName } from '@dsn/components-react/icon-registry.generated';
\`\`\`

Het bestand bestaat wel degelijk (`packages/components-react/src/Icon/icon-registry.generated.ts`), maar de `package.json` van `@dsn/components-react` declareert alleen een root export:

\`\`\`json
"exports": {
  ".": {
    "types": "./dist/index.d.ts",
    "import": "./dist/index.js",
    "default": "./dist/index.js"
  }
}
\`\`\`

Hierdoor gooit `tsc --noEmit` voor **11 story-bestanden** een TS2307-fout:

```
Cannot find module '@dsn/components-react/icon-registry.generated' or its corresponding type declarations.
```

**Fix**: subpath export toevoegen aan `packages/components-react/package.json`:

\`\`\`json
"exports": {
  ".": { ... },
  "./icon-registry.generated": {
    "types": "./dist/Icon/icon-registry.generated.d.ts",
    "import": "./dist/Icon/icon-registry.generated.js",
    "default": "./dist/Icon/icon-registry.generated.js"
  }
}
\`\`\`

De `dist/Icon/icon-registry.generated.*` bestanden worden al gegenereerd bij de build — dit maakt ze alleen formeel bereikbaar als subpath.

**Scope**: uitsluitend `packages/components-react/package.json` — één bestaand bestand, één toevoeging.

## Acceptance Criteria

- [ ] `packages/components-react/package.json` bevat een subpath export voor `./icon-registry.generated`
- [ ] `pnpm --filter storybook exec tsc --noEmit` geeft geen TS2307-fouten meer op `@dsn/components-react/icon-registry.generated` (was 11 fouten)
- [ ] De bestaande root export (`.`) is ongewijzigd

## Definition of Done

### Voorbereiding

- [ ] Feature branch aangemaakt: `git checkout -b fix/icon-registry-subpath-export`

### Werkzaamheden

- [ ] Implementatie afgerond

### Kwaliteitscontrole

- [ ] Tests groen: `pnpm test`
- [ ] TypeScript schoon: `pnpm --filter storybook exec tsc --noEmit`
- [ ] Lint schoon: `pnpm lint`

### Oplevering

- [ ] PR aangemaakt: `gh pr create`
- [ ] CI groen op de branch
- [ ] PR gemerged: `gh pr merge --merge`

## Notities / Open vragen

- Dit issue is afhankelijk van een succesvolle build (`pnpm build`) zodat de `dist/` bestanden aanwezig zijn — dit is al het geval in de huidige staat.
- Aangeraden om na #51 (MDX declarations) aan te pakken, zodat de overgebleven foutentelling per fix duidelijk is.

--- a/packages/components-react/package.json
+++ b/packages/components-react/package.json
@@ -10,6 +10,11 @@
       "types": "./dist/index.d.ts",
       "import": "./dist/index.js",
       "default": "./dist/index.js"
+    },
+    "./icon-registry.generated": {
+      "types": "./dist/Icon/icon-registry.generated.d.ts",
+      "import": "./dist/Icon/icon-registry.generated.js",
+      "default": "./dist/Icon/icon-registry.generated.js"
     }
   },
   "files": [

